utils/default.nix

```diff
@@ -18,13 +19,18 @@ let
     src = qmk-firmware-source;
     phases = [ "installPhase" ];
 
-    installPhase = ''
+    installPhase = let
+      target_dir = if type == "keyboard" then
+        "$out/keyboards"
+      else if type == "keymap" then
+        "$out/keyboards/${keyboard-name}/keymaps"
+      else throw "The only values valid for type are 'keyboard' and 'keymap'.";
```

Comment:
This is exactly why it would be nice using the modules system as I wrote [here](https://github.com/aciceri/qmk-nix-utils/pull/2#issuecomment-1669396598)
I swear I didn't read this PR yet when I commented that!

Reply:
I guess we are just getting more fluent in the Nix ecosystem!

We could either transform this into a module system interface or just start it afresh. I'd go for the second option since this repo, as it is right now, is alredy useful for whoever finds it. I definitely found it more ergonomic than the upstream `shell.nix`.